electric-operator.el

```diff
@@ -151,13 +176,16 @@ the given major mode."
         (cons "||" " || ")
         )
   "Default spacing rules for programming modes")
+(apply #'add-rules-for-mode 'electric-operator-prog-mode prog-mode-rules)
+
 
 (defvar prose-rules
   (add-rules '()
              (cons "." #'docs-.)
              (cons "," ", ")
              )
   "Rules to use in comments, strings and text modes.")
+(apply #'add-rules-for-mode 'electric-operator-prose-mode prose-rules)
```

Comment:
Hmm, I think this is a breaking change: previously users could modify `electric-operator-prose-rules` and the changes would be used, but now they can't. Hopefully no-one was doing that, but it's not a private variable so I think it would be a reasonable thing to do.

No one who has put their config on github is using it, but I don't know if that means much.

Any ideas?

Reply:
I figure we can do it pretty easily by compiling the regex on-the-fly if it isn't compiled. I've got something basic working for this but I need to play with it a bit more to get a clean implementation.

I'll merge this once I've got it figured out.

Reply:
I actually think prose-rules and prog-mode-rules should be made private and replaced by an appropriate entry in the hash-map. Currently they are the only two "modes" which aren't stored there, leading to an inconsistent interface when adding/changing rules as for all other modes the comment states that only the get/add-rules-for-modes functions should be used.